src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -311,17 +325,50 @@ populateJsonResponse(
     // no errors
     else if (result.txn)
     {
-        response = result.txn->getJson(JsonOptions::include_date, args.binary);
         auto const& sttx = result.txn->getSTransaction();
-        if (!args.binary)
-            RPC::insertDeliverMax(
-                response, sttx->getTxnType(), context.apiVersion);
+        if (context.apiVersion > 1)
+        {
+            std::string hash;
+            if (args.binary)
+                response[jss::tx_blob] = result.txn->getJson(
+                    JsonOptions::include_date, true, false, {std::ref(hash)});
+            else
+            {
+                response[jss::tx_json] = result.txn->getJson(
+                    JsonOptions::include_date, false, false, {std::ref(hash)});
+                RPC::insertDeliverMax(
+                    response[jss::tx_json],
+                    sttx->getTxnType(),
+                    context.apiVersion);
+            }
+            response[jss::hash] = hash;
+            if (result.validated)
+            {
+                response[jss::ledger_index] = result.txn->getLedger();
+                if (result.ledgerHash)
+                    response[jss::ledger_hash] = to_string(*result.ledgerHash);
+
+                if (result.closeTime)
```

Comment:
I think `response[jss::validated] = true;` can be added here. This is with reference to [Recommendations](https://github.com/XRPLF/rippled/issues/4727) point-7 

I notice that you are adding this field in `TransactionEntry.cpp` and `fillJsonTx`,  but I'm not sure if that's covering all possibilities.

Reply:
No, we set this field some 40 lines below
```
response[jss::validated] = result.validated;
```